rainmaker_orchestrator/server.py

```
@@ -1,426 +1,628 @@
-                    "modified": stat.st_mtime
-                })
-
-        return jsonify({
-            "files": files,
-            "count": len(files),
-            "workspace": workspace
-        }), 200
+        file_path = workspace_path / file.filename
+
+        if not file_path.is_relative_to(settings.workspace_path):
+            raise HTTPException(status_code=403, detail="Access denied")
```

Ruff (0.14.10)

Abstract `raise` to an inner function

(TRY301)